packages/core/database/src/types/index.ts

```diff
@@ -221,4 +223,8 @@ export interface Model {
   indexes?: Index[];
   foreignKeys?: ForeignKey[];
   lifecycles?: Partial<Record<Action, SubscriberFn>>;
+  options?: Record<string, unknown>;
+  modelName?: string;
+  // Should we import ModelType from @strapi/types?
+  modelType?: 'contentType' | 'component';
```

Comment:
@alexandrebodin I don't think the modelType is something that belongs to the database, but I didn't find any other way to check if the model was a content type. Do you know of any other alternative?

Reply:
Indeed. if you liik in the db there is a utility to transfor the models passed in with a todo to move it outside the DB. I think this should do the logic of adding Id / docId etc before it is even sent to the DB layer so you could avoid that and better separate concerns

Reply:
We will address this in another PR that will refactor some database types. If we try to do it here, it will involve a lot of unrelated decisions about where the types should go, if they are exposed to users, etc so it will be cleaner to do it later (or maybe even in v4 main and then update this to follow)